src/parser/src/first_pass/prop_controller.rs

```diff
@@ -119,6 +132,19 @@ impl PropController {
                 });
                 someid += 1;
             }
+            if let Some(wanted_state) = self.wanted_prop_states.get(&(bn.to_string())) {
+                self.wanted_prop_state_infos.push(WantedPropStateInfo {
+                    base: PropInfo {
+                        id: someid2,
+                        prop_type: PropType::Button,
+                        prop_name: bn.to_string(),
+                        prop_friendly_name: bn.to_string(),
+                        is_player_prop: true,
+                    },
+                    wanted_prop_state: wanted_state.clone(),
+                });
+                someid2 += 1;
```

Comment:
should i be creating a new ID (`someid2`) here or reusing `someid` is fine?

Reply:
You should be using the same one for it to work.

Reply:
Would it be ok then if the id is incremented twice for the same button? In the case if that button is in both - `wanted_player_props` and `wanted_prop_state_infos`? Or incremented once, but in the case that it was not in `wanted_player_props` and was in `wanted_prop_state_infos`?